src/database.js

```diff
@@ -36,7 +36,10 @@ function channel (path, event = 'value') {
   const channel = eventChannel(emit => {
     const callback = ref.on(
       event,
-      dataSnapshot => emit(dataSnapshot.val())
+      dataSnapshot => emit({
+        snapshot: dataSnapshot,
+        value: dataSnapshot.val()
```

Comment:
I renamed this in my local project to `data` instead of `value`, to keep consistency with Firebase doc examples. It's up to you though.